contributors/design-proposals/daemonset-update.md

```diff
@@ -158,9 +173,10 @@ For each pending DaemonSet updates, it will:
 1. Check `DaemonSetUpdateStrategy`:
    - If `OnDelete`: do nothing
    - If `RollingUpdate`:
-     - Compare spec of the daemon pods from step 1 with DaemonSet
-       `.spec.template.spec` to see if DaemonSet spec has changed.
-     - If DaemonSet spec has changed, compare `MaxUnavailable` with DaemonSet
+     - Find all daemon pods that belong to this DaemonSet using label selectors.
+       - Pods with "pod-template-generation" value equal to the DaemonSet's
```

Comment:
> I think kicking off a rollout just to relabel the pods once users switch to use the Rolling strategy is fine as long as we have the behavior documented.

I'm +1 for kicking off a rollout (killing them actively), since they won't be surprised by default (they have to modify update strategy to trigger this).